deploy-docker/src/index.ts

```diff
@@ -0,0 +1,90 @@
+import { promises as fs } from 'fs'
+import { getInput, setFailed } from '@actions/core'
+import { getOctokit } from '@actions/github'
+import { getBooleanInput } from '../../lib/core'
+import {
+  getShaFromRef,
+  getToken,
+  getRepository,
+  getRef,
+} from '../../lib/github'
+import { downloadS3object, getAWSCredentials } from '../../lib/s3'
+import { exec } from '@actions/exec'
+
+const METADATA_FILENAME = 'metadata.json'
+
+async function run() {
+  const { component: defaultComponent } = getRepository()
+  const component = getInput('component') || defaultComponent
+  const dockerUser = getInput('docker-user', { required: true })
+  const dockerPassword = getInput('docker-password', { required: true })
+  const dryRun = getBooleanInput('dry-run', false)
+  const dockerfile = getInput('dockerfile') || './Dockerfile'
+  const ghToken = getToken()
+
+  const [awsKeyId, awsSecretKey] = getAWSCredentials()
+
+  let imageVersion = getRef().split('/').reverse()[0]
+  if (getRef() === 'master' || getRef() === 'main') {
+    imageVersion = 'latest'
+  } else if (getRef() === 'develop') {
+    imageVersion = 'next'
+  }
+  console.log(`Image version will be: ${imageVersion}`)
+
+  const componentVersion =
+    process.env['GITHUB_REF']?.slice(0, 10) == 'refs/tags/'
+      ? process.env['GITHUB_REF']?.slice(10)
+      : process.env['GITHUB_SHA']
+  console.log(`Component version will be: ${componentVersion}`)
+
+  console.log(`Writing metadata to ${METADATA_FILENAME}`)
+  await fs.writeFile(
+    './' + METADATA_FILENAME,
+    JSON.stringify({ version: componentVersion })
+  )
+
+  const sha = await getShaFromRef({
+    octokit: getOctokit(ghToken),
+    component: 'env-to-config',
+    ref: 'main',
+  })
+
+  console.log('Downloading env-to-config')
+  await downloadS3object({
+    component: 'env-to-config',
+    sha,
+    usePlatformPrefix: true,
+    prefix: undefined,
+    path: `.`,
+    awsKeyId,
+    awsSecretKey,
+  })
```

Comment:
Could we add this to the Exivity base/merlin image instead of adding it here?
Appears to me it would be more predictable.